week-1-ts-ecosystem/w01e04-light-me-up/src/task1.ts

```diff
@@ -0,0 +1,18 @@
+import {PowerSource} from './models/powerSource';
+import {LightBulb} from './models/lightBulb';
+
+export default function task1() {
+  console.log("-- TASK 1 --");
+  const powerSource = new PowerSource();
+  console.log(`Starting with power: ${powerSource.getEnergyLeft()}`);
+  for (let i = 1; i <= 5; i++) {
+    new LightBulb(powerSource).consume();
+    console.log(`${i} light bulb, energy left: ${powerSource.getEnergyLeft()}`);
+  }
+
+  try {
+    new LightBulb(powerSource).consume();
+  } catch (e: any) {
+    console.log(e.message);
+  }
```

Comment:
Z czysto strategicznego punktu widzenia - jest ok.

Jednak zauważ że wkodzie produkcyjnym wszystkie odniesienia do `new LightBulb(powerSource).consume();` powinny być otoczone `try catch` więc ten `try` powinien się zaczynać w linii `:8`

Tutaj jak gdyby Ty - wiesz - że to się wywali przy `6` żarówce - dlatego się przygotowałeś do obsługi błędu przy `6`.
W rzeczywistym kodzie produkcyjnym "nie wiesz kiedy coś się wywali" ale możesz wiedzieć "że ma prawo się wywalić"
Dlatego otaczas to `try... catch`